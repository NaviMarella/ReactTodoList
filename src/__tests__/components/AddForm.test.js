import { shallow } from 'enzyme';
import React from 'react';
import TodoForm from './../../components/AddForm';

describe('Input Component', () => {
  test('render snapshot', () => {
    let props = {
        direction: "row",
        placeholderText: "Placeholder",
        maxLength: 10,
        addTODO: jest.fn()
    }

    const event = {
        preventDefault: jest.fn(),
        target: { value: 'text' }
    };
    const wrapper = shallow(<TodoForm {...props} />);
    wrapper.setState({ todoText: "test" })
    wrapper.instance().handleSubmit();
    expect(props.addTODO).toHaveBeenCalled();
    expect(wrapper).toMatchSnapshot();
  });

  test('render snapshot with column direction', () => {
      let props = {
        direction: "column",
        placeholderText: "Placeholder",
        maxLength: 10,
        addTODO: jest.fn()
      }
    const wrapper = shallow(<TodoForm {...props} />);
    wrapper.setState({
      todoText: "test"
    })
    wrapper.instance().handleInputChange("");
    expect(wrapper.state('todoText')).toEqual("")
    expect(wrapper).toMatchSnapshot();
  });

  test('render snapshot with empty input state', () => {
      let props = {
        direction: "column",
        placeholderText: "Placeholder",
        maxLength: 10,
        addTODO: jest.fn()
      }
    const wrapper = shallow(<TodoForm {...props} />);
    wrapper.setState({ todoText: "" })
    wrapper.instance().handleSubmit();
    expect(props.addTODO).toHaveBeenCalledTimes(0);
    expect(wrapper).toMatchSnapshot();
  });
});
