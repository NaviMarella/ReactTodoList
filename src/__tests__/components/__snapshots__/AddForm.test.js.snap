// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Input Component render snapshot 1`] = `
ShallowWrapper {
  Symbol(enzyme.__root__): [Circular],
  Symbol(enzyme.__unrendered__): <TodoForm
    addTODO={
      [MockFunction] {
        "calls": Array [
          Array [
            "test",
          ],
        ],
        "results": Array [
          Object {
            "isThrow": false,
            "value": undefined,
          },
        ],
      }
    }
    direction="row"
    maxLength={10}
    placeholderText="Placeholder"
  />,
  Symbol(enzyme.__renderer__): Object {
    "batchedUpdates": [Function],
    "getNode": [Function],
    "render": [Function],
    "simulateError": [Function],
    "simulateEvent": [Function],
    "unmount": [Function],
  },
  Symbol(enzyme.__node__): Object {
    "instance": null,
    "key": undefined,
    "nodeType": "function",
    "props": Object {
      "children": <form
        className="todo-form"
      >
        <Input
          handleInput={[Function]}
          maxLength={10}
          placeholder="Placeholder"
          type="search"
          value=""
        />
        <Button
          handleButtonClick={[Function]}
          type="submit"
        >
            Add 
        </Button>
      </form>,
      "type": "large",
    },
    "ref": null,
    "rendered": Object {
      "instance": null,
      "key": undefined,
      "nodeType": "host",
      "props": Object {
        "children": Array [
          <Input
            handleInput={[Function]}
            maxLength={10}
            placeholder="Placeholder"
            type="search"
            value=""
          />,
          <Button
            handleButtonClick={[Function]}
            type="submit"
          >
              Add 
          </Button>,
        ],
        "className": "todo-form",
      },
      "ref": null,
      "rendered": Array [
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "class",
          "props": Object {
            "handleInput": [Function],
            "maxLength": 10,
            "placeholder": "Placeholder",
            "type": "search",
            "value": "",
          },
          "ref": null,
          "rendered": null,
          "type": [Function],
        },
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "function",
          "props": Object {
            "children": "  Add ",
            "handleButtonClick": [Function],
            "type": "submit",
          },
          "ref": null,
          "rendered": "  Add ",
          "type": [Function],
        },
      ],
      "type": "form",
    },
    "type": [Function],
  },
  Symbol(enzyme.__nodes__): Array [
    Object {
      "instance": null,
      "key": undefined,
      "nodeType": "function",
      "props": Object {
        "children": <form
          className="todo-form"
        >
          <Input
            handleInput={[Function]}
            maxLength={10}
            placeholder="Placeholder"
            type="search"
            value=""
          />
          <Button
            handleButtonClick={[Function]}
            type="submit"
          >
              Add 
          </Button>
        </form>,
        "type": "large",
      },
      "ref": null,
      "rendered": Object {
        "instance": null,
        "key": undefined,
        "nodeType": "host",
        "props": Object {
          "children": Array [
            <Input
              handleInput={[Function]}
              maxLength={10}
              placeholder="Placeholder"
              type="search"
              value=""
            />,
            <Button
              handleButtonClick={[Function]}
              type="submit"
            >
                Add 
            </Button>,
          ],
          "className": "todo-form",
        },
        "ref": null,
        "rendered": Array [
          Object {
            "instance": null,
            "key": undefined,
            "nodeType": "class",
            "props": Object {
              "handleInput": [Function],
              "maxLength": 10,
              "placeholder": "Placeholder",
              "type": "search",
              "value": "",
            },
            "ref": null,
            "rendered": null,
            "type": [Function],
          },
          Object {
            "instance": null,
            "key": undefined,
            "nodeType": "function",
            "props": Object {
              "children": "  Add ",
              "handleButtonClick": [Function],
              "type": "submit",
            },
            "ref": null,
            "rendered": "  Add ",
            "type": [Function],
          },
        ],
        "type": "form",
      },
      "type": [Function],
    },
  ],
  Symbol(enzyme.__options__): Object {
    "adapter": ReactSixteenAdapter {
      "options": Object {
        "enableComponentDidUpdateOnSetState": true,
        "lifecycles": Object {
          "componentDidUpdate": Object {
            "onSetState": true,
          },
          "getDerivedStateFromProps": true,
          "getSnapshotBeforeUpdate": true,
          "setState": Object {
            "skipsComponentDidUpdateOnNullish": true,
          },
        },
      },
    },
  },
}
`;

exports[`Input Component render snapshot with column direction 1`] = `
ShallowWrapper {
  Symbol(enzyme.__root__): [Circular],
  Symbol(enzyme.__unrendered__): <TodoForm
    addTODO={[MockFunction]}
    direction="column"
    maxLength={10}
    placeholderText="Placeholder"
  />,
  Symbol(enzyme.__renderer__): Object {
    "batchedUpdates": [Function],
    "getNode": [Function],
    "render": [Function],
    "simulateError": [Function],
    "simulateEvent": [Function],
    "unmount": [Function],
  },
  Symbol(enzyme.__node__): Object {
    "instance": null,
    "key": undefined,
    "nodeType": "function",
    "props": Object {
      "children": <form
        className="todo-form__column"
      >
        <Input
          handleInput={[Function]}
          maxLength={10}
          placeholder="Placeholder"
          type="search"
          value=""
        />
        <Button
          handleButtonClick={[Function]}
          type="submit"
        >
            Add 
        </Button>
      </form>,
      "type": "small",
    },
    "ref": null,
    "rendered": Object {
      "instance": null,
      "key": undefined,
      "nodeType": "host",
      "props": Object {
        "children": Array [
          <Input
            handleInput={[Function]}
            maxLength={10}
            placeholder="Placeholder"
            type="search"
            value=""
          />,
          <Button
            handleButtonClick={[Function]}
            type="submit"
          >
              Add 
          </Button>,
        ],
        "className": "todo-form__column",
      },
      "ref": null,
      "rendered": Array [
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "class",
          "props": Object {
            "handleInput": [Function],
            "maxLength": 10,
            "placeholder": "Placeholder",
            "type": "search",
            "value": "",
          },
          "ref": null,
          "rendered": null,
          "type": [Function],
        },
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "function",
          "props": Object {
            "children": "  Add ",
            "handleButtonClick": [Function],
            "type": "submit",
          },
          "ref": null,
          "rendered": "  Add ",
          "type": [Function],
        },
      ],
      "type": "form",
    },
    "type": [Function],
  },
  Symbol(enzyme.__nodes__): Array [
    Object {
      "instance": null,
      "key": undefined,
      "nodeType": "function",
      "props": Object {
        "children": <form
          className="todo-form__column"
        >
          <Input
            handleInput={[Function]}
            maxLength={10}
            placeholder="Placeholder"
            type="search"
            value=""
          />
          <Button
            handleButtonClick={[Function]}
            type="submit"
          >
              Add 
          </Button>
        </form>,
        "type": "small",
      },
      "ref": null,
      "rendered": Object {
        "instance": null,
        "key": undefined,
        "nodeType": "host",
        "props": Object {
          "children": Array [
            <Input
              handleInput={[Function]}
              maxLength={10}
              placeholder="Placeholder"
              type="search"
              value=""
            />,
            <Button
              handleButtonClick={[Function]}
              type="submit"
            >
                Add 
            </Button>,
          ],
          "className": "todo-form__column",
        },
        "ref": null,
        "rendered": Array [
          Object {
            "instance": null,
            "key": undefined,
            "nodeType": "class",
            "props": Object {
              "handleInput": [Function],
              "maxLength": 10,
              "placeholder": "Placeholder",
              "type": "search",
              "value": "",
            },
            "ref": null,
            "rendered": null,
            "type": [Function],
          },
          Object {
            "instance": null,
            "key": undefined,
            "nodeType": "function",
            "props": Object {
              "children": "  Add ",
              "handleButtonClick": [Function],
              "type": "submit",
            },
            "ref": null,
            "rendered": "  Add ",
            "type": [Function],
          },
        ],
        "type": "form",
      },
      "type": [Function],
    },
  ],
  Symbol(enzyme.__options__): Object {
    "adapter": ReactSixteenAdapter {
      "options": Object {
        "enableComponentDidUpdateOnSetState": true,
        "lifecycles": Object {
          "componentDidUpdate": Object {
            "onSetState": true,
          },
          "getDerivedStateFromProps": true,
          "getSnapshotBeforeUpdate": true,
          "setState": Object {
            "skipsComponentDidUpdateOnNullish": true,
          },
        },
      },
    },
  },
}
`;

exports[`Input Component render snapshot with empty input state 1`] = `
ShallowWrapper {
  Symbol(enzyme.__root__): [Circular],
  Symbol(enzyme.__unrendered__): <TodoForm
    addTODO={[MockFunction]}
    direction="column"
    maxLength={10}
    placeholderText="Placeholder"
  />,
  Symbol(enzyme.__renderer__): Object {
    "batchedUpdates": [Function],
    "getNode": [Function],
    "render": [Function],
    "simulateError": [Function],
    "simulateEvent": [Function],
    "unmount": [Function],
  },
  Symbol(enzyme.__node__): Object {
    "instance": null,
    "key": undefined,
    "nodeType": "function",
    "props": Object {
      "children": <form
        className="todo-form__column"
      >
        <Input
          handleInput={[Function]}
          maxLength={10}
          placeholder="Placeholder"
          type="search"
          value=""
        />
        <Button
          handleButtonClick={[Function]}
          type="submit"
        >
            Add 
        </Button>
      </form>,
      "type": "small",
    },
    "ref": null,
    "rendered": Object {
      "instance": null,
      "key": undefined,
      "nodeType": "host",
      "props": Object {
        "children": Array [
          <Input
            handleInput={[Function]}
            maxLength={10}
            placeholder="Placeholder"
            type="search"
            value=""
          />,
          <Button
            handleButtonClick={[Function]}
            type="submit"
          >
              Add 
          </Button>,
        ],
        "className": "todo-form__column",
      },
      "ref": null,
      "rendered": Array [
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "class",
          "props": Object {
            "handleInput": [Function],
            "maxLength": 10,
            "placeholder": "Placeholder",
            "type": "search",
            "value": "",
          },
          "ref": null,
          "rendered": null,
          "type": [Function],
        },
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "function",
          "props": Object {
            "children": "  Add ",
            "handleButtonClick": [Function],
            "type": "submit",
          },
          "ref": null,
          "rendered": "  Add ",
          "type": [Function],
        },
      ],
      "type": "form",
    },
    "type": [Function],
  },
  Symbol(enzyme.__nodes__): Array [
    Object {
      "instance": null,
      "key": undefined,
      "nodeType": "function",
      "props": Object {
        "children": <form
          className="todo-form__column"
        >
          <Input
            handleInput={[Function]}
            maxLength={10}
            placeholder="Placeholder"
            type="search"
            value=""
          />
          <Button
            handleButtonClick={[Function]}
            type="submit"
          >
              Add 
          </Button>
        </form>,
        "type": "small",
      },
      "ref": null,
      "rendered": Object {
        "instance": null,
        "key": undefined,
        "nodeType": "host",
        "props": Object {
          "children": Array [
            <Input
              handleInput={[Function]}
              maxLength={10}
              placeholder="Placeholder"
              type="search"
              value=""
            />,
            <Button
              handleButtonClick={[Function]}
              type="submit"
            >
                Add 
            </Button>,
          ],
          "className": "todo-form__column",
        },
        "ref": null,
        "rendered": Array [
          Object {
            "instance": null,
            "key": undefined,
            "nodeType": "class",
            "props": Object {
              "handleInput": [Function],
              "maxLength": 10,
              "placeholder": "Placeholder",
              "type": "search",
              "value": "",
            },
            "ref": null,
            "rendered": null,
            "type": [Function],
          },
          Object {
            "instance": null,
            "key": undefined,
            "nodeType": "function",
            "props": Object {
              "children": "  Add ",
              "handleButtonClick": [Function],
              "type": "submit",
            },
            "ref": null,
            "rendered": "  Add ",
            "type": [Function],
          },
        ],
        "type": "form",
      },
      "type": [Function],
    },
  ],
  Symbol(enzyme.__options__): Object {
    "adapter": ReactSixteenAdapter {
      "options": Object {
        "enableComponentDidUpdateOnSetState": true,
        "lifecycles": Object {
          "componentDidUpdate": Object {
            "onSetState": true,
          },
          "getDerivedStateFromProps": true,
          "getSnapshotBeforeUpdate": true,
          "setState": Object {
            "skipsComponentDidUpdateOnNullish": true,
          },
        },
      },
    },
  },
}
`;
